File: app/src/main/java/com/sl/databindingdemo/inversemethod/InverseMethodUtil.java
package com.sl.databindingdemo.inversemethod;

import androidx.databinding.InverseMethod;

public class InverseMethodUtil {
    @InverseMethod("convertIntToString")
    public static int convertStringToInt(String value) {
        try {
            return Integer.parseInt(value);
        } catch (NumberFormatException e) {
            return -1;
        }
    }

    public static String convertIntToString(int value) {
        return String.valueOf(value);
    }
}

File: app/src/main/java/com/sl/databindingdemo/inversemethod/InverseMethodUtil.kt
package com.sl.databindingdemo.inversemethod

import androidx.databinding.InverseMethod
import java.lang.Exception

@InverseMethod("stringToInt")
fun intToString(i: Int?): String? {
    return i?.toString()
}

fun stringToInt(str: String?): Int? {
    if (str == null){
        return null
    }
    try {
        return str.toInt()
    }catch (e : Exception){

    }
    return null

}
